pysat/tests/test_constellation.py

```diff
@@ -577,3 +575,91 @@ def test_to_inst_mult_pad_clean(self):
                                    list(self.inst[1].index))
 
         return
+
+    @pytest.mark.parametrize("method", ["del", "drop"])
+    def test_delitem_all_inst(self, method):
+        """Test Constellation deletion of data variables from all instruments.
+
+        Parameters
+        ----------
+        method : str
+            String specifying the deletion method
+
+        """
+        # Load the Constellation data
+        self.const.load(date=self.ref_time)
+
+        # Delete the UTS data from all instruments
+        dvar = "uts"
+        if method == "del":
+            del self.const[dvar]
+        else:
+            self.const.drop(dvar)
+
+        # Test that this variable is gone from all Instruments
+        for inst in self.const.instruments:
+            assert dvar not in inst.variables
+
+        # Test that the constellation variable list has been updated
+        for var in self.const.variables:
+            assert var.find(dvar) != 0
+
+        return
+
+    @pytest.mark.parametrize("method", ["del", "drop"])
+    def test_delitem_one_inst(self, method):
+        """Test Constellation deletion of data variables from one instrument.
+
+        Parameters
+        ----------
+        method : str
+            String specifying the deletion method
+
+        """
+        # Load the Constellation data
+        self.const.load(date=self.ref_time)
+
+        # Delete the UTS data from all instruments
+        dvar = "uts_pysat_testing"
+        if method == "del":
+            del self.const[dvar]
+        else:
+            self.const.drop(dvar)
+
+        # Test that this variable is gone from only the desired Instrument
+        for inst in self.const.instruments:
+            if inst.platform == "pysat" and inst.name == "testing":
```

Comment:
Good point, better to remove the if/else and keep the drop test condition.

Reply:
This is used. The class pulls from the `constellations.testing` object, which means the constellation should include three instruments: `pysat_testing`, `pysat_ndtesting`, and `pysat_testmodel`.

Reply:
It's been so long since I wrote that 🐟

Reply:
This whole section underwent a major update since we dropped a bunch of deprecated instruments, so it's good to double-check everything. Took me a while to find it.